internal/querycoordv2/observers/collection_observer.go

```diff
@@ -98,51 +117,104 @@ func (ob *CollectionObserver) Stop() {
 	})
 }
 
+func (ob *CollectionObserver) LoadCollection(ctx context.Context, collectionID int64) {
+	span := trace.SpanFromContext(ctx)
+
+	traceID := span.SpanContext().TraceID()
+	key := traceID.String()
+
+	if !traceID.IsValid() {
+		key = fmt.Sprintf("LoadCollection_%d", collectionID)
+	}
+
+	ob.loadTasks.Insert(key, LoadTask{LoadType: querypb.LoadType_LoadCollection, CollectionID: collectionID})
```

Comment:
could this be a duplicated task?
what happens if user load collection and then load partitions or vice versa?

Reply:
I think load collection can overwrite load partitions. 
But we happen if load partitions after  load collection?

Reply:
@xiaofan-luan for load partition after load collection, only `lacked partition` will be added as a load task which shall not happen since `SyncNewCreatedPartition` API will add partition into meta first